### ISSUE TITLE
belongs_to input has a different id to the label

### ISSUE THREAD

**Issue by iainbeeston** [avo-hq/avo#1214]
### Describe the bug
If you have a belongs_to field, the `for` attribute of the label does not match the `id` of the `select` menu. This causes accessibility issues including clicking on the label does not focus the field and capybara cannot find the input associated with the label.

I've tried to figure out what the issue is exactly with the belongs to field but I haven't been able to figure it out.
 
### Steps to Reproduce
1. Add a belongs to field to a resource (for example the Membership resource in the demo app)
2. Go to the edit page for an instance of that resource ([demo app](https://avodemo.herokuapp.com/avo/resources/memberships/43/edit?via_resource_class=Team&via_resource_id=4))
3. Click on the label

### Expected behavior & Actual behavior
The select menu for that field should gain focus, but it doesn't.

### System configuration
**Avo version**: 2.14.2

**Rails version**: 7.0.3.1

**Ruby version**: 3.1.2

**License type**:

 - [ ] Community
 - [x] Pro

#### Impact

 - [ ] High impact
 - [ ] Medium impact
 - [x] Low impact

#### Urgency

 - [ ] High urgency
 - [ ] Medium urgency
 - [x] low urgency

**Comment by adrianthedev** [comment]
This has been fixed.
